### PR TITLE
fix: Update Hello World template to match the ones in the templates repository

### PR DESCRIPTION
**Pull request by jonife**
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by mgrandis — APPROVED** [review]

File: samcli/lib/init/templates/cookiecutter-aws-sam-hello-dotnet/cookiecutter.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
     "project_name": "Name of the project",
-    "runtime": "dotnetcore2.1"
+    "runtime": "dotnetcore3.1"
```
Comment (mndeveci) [inline samcli/lib/init/templates/cookiecutter-aws-sam-hello-dotnet/cookiecutter.json:3]: Should we keep this as `dotnetcore2.1` for backwards compatibility?

File: samcli/lib/init/templates/cookiecutter-aws-sam-hello-nodejs/cookiecutter.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
     "project_name": "Name of the project",
-    "runtime": "nodejs12.x"
+    "runtime": "nodejs14.x"
```
Comment (mndeveci) [inline samcli/lib/init/templates/cookiecutter-aws-sam-hello-nodejs/cookiecutter.json:3]: Same as the other one, should we keep original version inside SAM CLI?

**Review by mndeveci — COMMENTED** [review]
Thanks for syncing this with external repo!

It looks good in general, I have questions about bumping dotnetcore and nodejs versions

File: samcli/lib/init/templates/cookiecutter-aws-sam-hello-dotnet/cookiecutter.json (hunk 1 above)
Comment (jfuss) [reply samcli/lib/init/templates/cookiecutter-aws-sam-hello-dotnet/cookiecutter.json:3]: `dotnetcore2.1` is no longer supported by Lambda. https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html

File: samcli/lib/init/templates/cookiecutter-aws-sam-hello-nodejs/cookiecutter.json (hunk 2 above)
Comment (jonife) [reply samcli/lib/init/templates/cookiecutter-aws-sam-hello-nodejs/cookiecutter.json:3]: So with the current INIT design we default to the latest version of the runtime selected by the customer for hello world templates however customers can still specify the runtime version they would like to use. So defaulting to node14.x as against nodejs12.x was necessary and after working on this project I don't see how this affects backward compatibility.

**Pull request closed without merging**